File: ios/PizzaDelivery/Sources/Models/Recommendation.swift
import Foundation

struct Recommendation: Codable, Identifiable {
    let id: String
    var title: String
    var description: String?
    var items: [MenuItem]?
    var type: RecommendationType
}

enum RecommendationType: String, Codable {
    case personalized
    case trending
    case newItems = "new_items"
    case reorder
}

struct Offer: Codable, Identifiable {
    let id: String
    var title: String
    var description: String
    var code: String
    var discount: Double
    var discountType: String
    var minOrder: Double?
    var validUntil: String?
    var imageUrl: String?
}

File: android/app/src/main/java/com/pizzadelivery/android/data/model/Recommendation.kt
package com.pizzadelivery.android.data.model

data class Recommendation(
    val id: String = "",
    val title: String = "",
    val description: String = "",
    val items: List<MenuItem> = emptyList(),
    val type: RecommendationType = RecommendationType.PERSONALIZED
)

enum class RecommendationType {
    PERSONALIZED,
    TRENDING,
    NEW_ARRIVALS,
    OFFERS
}

data class Offer(
    val id: String = "",
    val title: String = "",
    val description: String = "",
    val code: String = "",
    val discount: Double = 0.0,
    val discountType: String = "percentage",
    val image: String = "",
    val validUntil: String = ""
)
